ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/MergeTransitionBlockValidator.java

```diff
@@ -139,6 +140,9 @@ private SafeFuture<PayloadValidationResult> retrieveAndVerifyNonFinalizedTransit
 
   private SafeFuture<PayloadStatus> verifyTransitionPayload(
       final UInt64 slot, final ExecutionPayloadSummary executionPayload) {
+    if (spec.atSlot(slot).getMilestone().isGreaterThanOrEqualTo(SpecMilestone.CAPELLA)) {
```

Comment:
Problem i see is this is called in `verifyFinalizedTransitionBlock` where 
```
final Optional<SlotAndExecutionPayloadSummary> finalizedOptimisticTransitionPayload =
        recentChainData.getStore().getFinalizedOptimisticTransitionPayload();
```

The slot retrieved would always be bellatrix one for mainnet and we would still end up calling `validateMergeBlock()` so spec wouldn't be met.

Reply:
I'm not sure if it was removed in Capella. Actually I'm a bit confused to understand what part of spec it resembles. The shortcut was made for Devnets with genesis starting from Capella or later forks, not targeting Mainnet